core/templates/dev/head/pages/exploration-editor-page/services/exploration-states.service.ts

```diff
@@ -382,6 +401,12 @@ oppia.factory('ExplorationStatesService', [
         saveStateProperty(
           stateName, 'written_translations', newWrittenTranslations);
       },
+      saveImage: function(stateName, image) {
+        addImageInState(stateName, image);
+      },
+      getImageSource: function(stateName, imageId) {
```

Comment:
This seems more of a imageAssets property. I'm not sure how this method is integrated, can you give a brief info on when and how it's get called?

Reply:
since ```ExplorationStatesService``` is responsible for state-related data in the editor. 
So whenever there is an addition of image, after adding image source in the state, the image source is extracted back with the help of this function by passing imageId and stateName.